packages/service-worker/src/services/keystone/handlers/migrateMissingKeysFromKeystone.ts

```diff
@@ -0,0 +1,181 @@
+import { getAddressDerivationPath } from '@avalabs/core-wallets-sdk';
+import {
+  ExtensionRequest,
+  ExtensionRequestHandler,
+  AddressPublicKeyJson,
+  SecretType,
+  ExtendedPublicKey,
+} from '@core/types';
+import { injectable } from 'tsyringe';
+import { SecretsService } from '../../secrets/SecretsService';
+import { AccountsService } from '../../accounts/AccountsService';
+import {
+  buildExtendedPublicKey,
+  getExtendedPublicKey,
+  hasPublicKeyFor,
+} from '../../secrets/utils';
+import { getAvalancheExtendedKeyPath } from '@core/common';
+import { AddressPublicKey } from '~/services/secrets/AddressPublicKey';
+
+import KeystoneUSBAvalancheSDK from '@keystonehq/hw-app-avalanche';
+import { createKeystoneTransport } from '@keystonehq/hw-transport-webusb';
+import {
+  Curve as KeystoneCurve,
+  DerivationAlgorithm,
+} from '@keystonehq/bc-ur-registry';
+import { fromPublicKey } from 'bip32';
+
+type HandlerType = ExtensionRequestHandler<
+  ExtensionRequest.KEYSTONE_MIGRATE_MISSING_PUBKEYS,
+  boolean
+>;
+
+@injectable()
+export class MigrateMissingPublicKeysFromKeystoneHandler
+  implements HandlerType
+{
+  method = ExtensionRequest.KEYSTONE_MIGRATE_MISSING_PUBKEYS as const;
+
+  constructor(
+    private secretsService: SecretsService,
+    private accountsService: AccountsService,
+  ) {}
+
+  handle: HandlerType['handle'] = async ({ request }) => {
+    try {
+      const activeAccount = await this.accountsService.getActiveAccount();
+      if (!activeAccount) {
+        throw new Error('There is no active account');
+      }
+      const secrets =
+        await this.secretsService.getAccountSecrets(activeAccount);
+
+      if (secrets.secretType !== SecretType.Keystone3Pro) {
```

Comment:
Do we also need to check that the currently active account is for the same Keystone? What happens if the user has 3 devices